### PR TITLE
Nz/3864 joi outcomes metrics

### LINKED ISSUE #3864: [Feature] Update Outcomes and Metrics form to use new tools
**Description and related issues -**
Update the Outcomes and Metrics form to use react-hook-form and joi. See #3499 for details on how to implement it.

### This task is done when…
- [ ] the form is being validated using the tools above

### PR DESCRIPTION
**Pull request by Sun-Mountain**
Resolves #3864 

**Description-**
Update the Outcomes and Metrics form to use react-hook-form and joi. See [#3499](https://github.com/CMSgov/eAPD/issues/3499) for details on how to implement it.

**This pull request changes...**

- Outcome and metric form will not allow save unless form is valid.

**This pull request also touches…**

- Uses joi testing-library linting.

**This pull request was tested in the follow ways…**

- list of frontend cases
- list of backend cases

**Steps to manually verify this change...**

1. steps to view and verify change

### This pull request is ready to review when...

- [x] Automated tests are updated (and all tests are passing)
- [x] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented

### This pull request can be merged when…

- [ ] Code has been reviewed by someone other than the original author
- [ ] QA has verified the accessibility and functionality related to the change
- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -712,7 +712,7 @@ jobs:
               --AWS_SECURITY_GROUP "${<< parameters.env >>_API_AWS_SECURITY_GROUP}" \
               --AWS_SUBNET "${<< parameters.env >>_API_AWS_SUBNET}" \
               --AWS_TARGET_GROUP "${<< parameters.env >>_API_AWS_TARGET_GROUP}" \
-              --BUILD_URL "$(cat ../../build-url.txt)/backend.zip" \
+              --BUILD_URL "$(cat ../../build-url.txt)" \
```
Comment (tbolt) [inline .circleci/config.yml:715]: Probably double check with @cthulhuplus if this is the latest

File: web/package.json [hunk 2]
```diff
@@ -6,7 +6,6 @@
   "scripts": {
     "audit": "npx audit-ci --config ./audit-ci.json",
     "make-state-icons": "babel-node scripts/stateIcons.js && cp src/data/states-small/* src/static/img/states/",
-    "copy-schemas": "cp -r ../schemas src/static",
```
Comment (tbolt) [inline web/package.json:9]: Does Tif want to get rid of this?

File: web/src/containers/activity/ContractorResource/ContractorResourceForm.test.js [hunk 3]
```diff
@@ -17,15 +17,17 @@ const defaultProps = {
...
-    useHourly: false,
     hourly: {
-      1066: {
-        hours: 10,
-        rate: 100
-      },
-      1067: {
-        hours: 20,
-        rate: 200
+      useHourly: false,
```
Comment (tbolt) [inline web/src/containers/activity/ContractorResource/ContractorResourceForm.test.js:21]: How Tif has it in her branch is the way it should be based on the new schema

File: web/src/containers/activity/OutcomesAndMetrics/OutcomeAndMetricForm.js [hunk 4]
```diff
@@ -39,6 +65,9 @@ const OutcomeAndMetricForm = forwardRef(
         }
         case 'updateMetrics': {
           const metricsCopy = [...state.metrics];
+          console.log({ metricsCopy });
```
Comment (tbolt) [inline web/src/containers/activity/OutcomesAndMetrics/OutcomeAndMetricForm.js:68]: console.logs

File: web/src/containers/activity/OutcomesAndMetrics/OutcomeAndMetricForm.js [hunk 5]
```diff
@@ -53,91 +82,127 @@ const OutcomeAndMetricForm = forwardRef(
...
-            onDeleteLabel="Remove"
-            skipConfirmation
-            ariaLabel={`${i + 1}. ${metric || 'Metric not specified'}`}
-            objType="Metric"
-          >
-            <div
-              key={key}
-              className="ds-c-choice__checkedChild ds-u-margin-top--3 ds-u-padding-top--0"
+        {fields.map((metric, i) => {
+          console.log({ metric, index });
```
Comment (tbolt) [inline web/src/containers/activity/OutcomesAndMetrics/OutcomeAndMetricForm.js:147]: console.log

**Review by tbolt — CHANGES_REQUESTED** [review]
Code looks good. Left comments on some minor housekeeping things. I'll msg you on some clarifications around testing.

File: web/package.json (hunk 2 above)
Comment (thetif) [reply web/package.json:9]: we will when I get done with this workspace stuff

**Review by thetif — APPROVED** [review]

**Review by thetif — APPROVED** [review]
approved when you fix the small suggestions Ty has

File: web/package.json (hunk 2 above)
Comment (Sun-Mountain) [reply web/package.json:9]: Should I leave as is (deleted in my branch) or bring it back?

**Review by mirano-darren — APPROVED** [review]
Looks good to me, well done!

**Review by tbolt — APPROVED** [review]

**Review by jeromeleecms — APPROVED** [review]
Looks good! Whew!

**Pull request merged**